managed/services/management/node_test.go

```diff
@@ -148,4 +153,303 @@ func TestNodeService(t *testing.T) {
 			})
 		})
 	})
+
+	t.Run("ListNodes", func(t *testing.T) {
+		now = models.Now()
+
+		setup := func(t *testing.T) (ctx context.Context, s *MgmtNodeService, teardown func(t *testing.T)) {
+			t.Helper()
+
+			origNowF := models.Now
+			models.Now = func() time.Time {
+				return now
+			}
+
+			ctx = logger.Set(context.Background(), t.Name())
+			uuid.SetRand(&tests.IDReader{})
+
+			sqlDB := testdb.Open(t, models.SetupFixtures, nil)
+			db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+
+			ar := &mockAgentsRegistry{}
+			ar.Test(t)
+
+			vmClient := &mockVictoriaMetricsClient{}
+			vmClient.Test(t)
+
+			s = NewMgmtNodeService(db, ar, vmClient)
+
+			teardown = func(t *testing.T) {
+				models.Now = origNowF
+				uuid.SetRand(nil)
+
+				require.NoError(t, sqlDB.Close())
+				ar.AssertExpectations(t)
+			}
+
+			return
+		}
+
+		const (
+			nodeExporterID      = "/agent_id/00000000-0000-4000-8000-000000000001"
+			postgresqlServiceID = "/service_id/00000000-0000-4000-8000-000000000002"
+		)
+
+		t.Run("should output an unfiltered list of all nodes", func(t *testing.T) {
+			ctx, s, teardown := setup(t)
+			defer teardown(t)
+
+			metric := model.Vector{
+				&model.Sample{
+					Metric: model.Metric{
+						"__name__": "up",
+						"node_id":  "pmm-server",
+					},
+					Timestamp: 1,
+					Value:     1,
+				},
+			}
+
+			s.vmClient.(*mockVictoriaMetricsClient).On("Query", ctx, mock.Anything, mock.Anything).Return(metric, nil, nil).Times(2)
+			s.r.(*mockAgentsRegistry).On("IsConnected", models.PMMServerAgentID).Return(true).Once()
+			s.r.(*mockAgentsRegistry).On("IsConnected", nodeExporterID).Return(true).Once()
+			res, err := s.ListNodes(ctx, &nodev1beta1.ListNodeRequest{})
+
+			expected := &nodev1beta1.ListNodeResponse{
+				Nodes: []*nodev1beta1.UniversalNode{
+					{
+						NodeId:        "pmm-server",
+						NodeType:      "generic",
+						NodeName:      "pmm-server",
+						MachineId:     "",
+						Distro:        "",
+						NodeModel:     "",
+						ContainerId:   "",
+						ContainerName: "",
+						Address:       "127.0.0.1",
+						Region:        "",
+						Az:            "",
+						CustomLabels:  nil,
+						CreatedAt:     timestamppb.New(now),
+						UpdatedAt:     timestamppb.New(now),
+						Agents: []*nodev1beta1.UniversalNode_Agent{
+							{
+								AgentId:     nodeExporterID,
+								AgentType:   "node_exporter",
+								Status:      "UNKNOWN",
+								IsConnected: true,
+							},
+							{
+								AgentId:     models.PMMServerAgentID,
+								AgentType:   "pmm-agent",
+								Status:      "",
+								IsConnected: true,
+							},
+						},
+						Services: []*nodev1beta1.UniversalNode_Service{
+							{
+								ServiceId:   postgresqlServiceID,
+								ServiceType: "postgresql",
+								ServiceName: "pmm-server-postgresql",
+							},
+						},
+						Status: nodev1beta1.UniversalNode_UP,
+					},
+				},
+			}
+
+			assert.NoError(t, err)
```

Comment:
It's better to place assertion right after function call.

Reply:
Done.